Code/src/main/java/nl/utwente/group10/ui/gestures/InputAnchorHandler.java

```diff
@@ -0,0 +1,129 @@
+				if (mdEvent.getEventType().equals(
+						MouseDragEvent.MOUSE_DRAG_RELEASED)) {
+					if (mdEvent.getGestureSource() instanceof OutputAnchor) {
+						// Finalize connection
+						if (!inputAnchor.getConnection().isPresent()) {
+							createConnection(mdEvent);
+						}
+					}
+				}
+			} else if (mEvent.getEventType().equals(MouseEvent.DRAG_DETECTED)) {
+				if (inputAnchor.getConnection().isPresent()) {
+					// Edit (move) existing connection
+					moveConnection(MOUSE_ID);
+				}
+			} else if (mEvent.getEventType().equals(MouseEvent.MOUSE_DRAGGED)) {
+				// Move in progress visual-only connection (follow cursor)
+				if (lines.get(MOUSE_ID) != null) {
+					updateLine(MOUSE_ID, mEvent.getSceneX(), mEvent.getSceneY());
+				}
+			} else if (mEvent.getEventType().equals(MouseEvent.MOUSE_RELEASED)) {
+				// Remove visual-only connection
+				if (lines.get(MOUSE_ID) != null) {
+					finalizeLine(MOUSE_ID);
+				}
+			}
+		} else if (event instanceof TouchEvent) {
+			TouchEvent tEvent = ((TouchEvent) event);
+			int touchID = tEvent.getTouchPoint().getId();
+			if (tEvent.getEventType().equals(TouchEvent.TOUCH_PRESSED)) {
+				moveConnection(touchID);
+			} else if (tEvent.getEventType().equals(TouchEvent.TOUCH_MOVED)) {
+				if (lines.get(touchID) != null) {
+					updateLine(touchID, tEvent.getTouchPoint().getSceneX(),
+							tEvent.getTouchPoint().getSceneY());
+				}
+			} else if (tEvent.getEventType().equals(TouchEvent.TOUCH_RELEASED)) {
+				if (lines.get(touchID) != null) {
+					finalizeLine(touchID);
+				}
+			}
+		}
+
+		// TODO TouchDragEvent does not exist. Test if touch fires synthesized
+		// MouseDragEvents, else find workaround.
+
+		event.consume();
+	}
+
+	private void createLine(int id, double x, double y) {
+		ConnectionLine line = new ConnectionLine();
+		cpane.getChildren().add(line);
+		line.setMouseTransparent(true);
+		line.setStartPosition(x, y);
+		line.setEndPosition(x, y);
+		lines.put(id, line);
+	}
+
+	private void updateLine(int id, double x, double y) {
+		lines.get(id).setEndPosition(x, y);
+	}
+
+	private void finalizeLine(int id) {
+		lines.get(id).setMouseTransparent(false);
+		cpane.getChildren().remove(lines.get(id));
+		cpane.invalidate();
+		lines.remove(id);
+	}
+
+	private void removeConnection() {
+		Connection connection =inputAnchor.getConnection().get(); 
+		connection.disconnect();
+		cpane.getChildren().remove(connection);
+	}
+
+	private void createConnection(MouseDragEvent mdEvent) {
+		inputAnchor.createConnectionFrom((OutputAnchor) mdEvent
+				.getGestureSource());
+		cpane.invalidate();
+	}
+
+	private void moveConnection(int id) {
+		OutputAnchor outputAnchor = inputAnchor.getConnection().get()
+				.getOutputAnchor();
+		outputAnchor.startFullDrag();
+		Point2D point = outputAnchor.localToScene(outputAnchor.getCenterX(),
+				outputAnchor.getCenterY());
+		createLine(id, point.getX(), point.getY());
+		removeConnection();
+	}
+}
```

Comment:
Dit is wel in zekere mate nodig zodat je op meerdere plekken tegelijk lijnen kunt trekken. Ik vind het vooral niet zo mooi dat elke handler zijn eigen lijstje bijhoud. Hier komt mogelijk iets van een LineManager klasse tussen.